plugins/wazuh-core/server/services/initialization/routes.ts

```diff
@@ -0,0 +1,242 @@
+              .map(({ name }) => name)
+              .join(', ')}`,
+            tasks: results,
+          },
+        });
+      } catch (e) {
+        return response.internalError({
+          body: {
+            message: `Error running the internal initialization tasks: ${e.message}`,
+          },
+        });
+      }
+    },
+  );
+
+  router.post(
+    {
+      path: `${apiEndpointBase}/user`,
+      validate: {
+        // TODO: restrict to user tasks
+        query: schema.object({
+          tasks: validateTaskList,
+        }),
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const tasksNames = request.query.tasks
+          ? getTaskList(request.query.tasks)
+          : undefined;
+        const logger = context.wazuh_core.logger;
+        const username = ''; // TODO: get value
+        const scope = 'user';
+        logger.debug(
+          `Getting initialization tasks related to user [${username}] scope [${scope}]`,
+        );
+        const initializationTasks = context.wazuh_core.initialization.get();
+
+        const indexPatternTasks = initializationTasks
+          .filter(({ name }) => name.startsWith('index-pattern:'))
+          .map(({ name }) =>
+            context.wazuh_core.initialization.createNewTaskFromRegisteredTask(
+              name,
+            ),
+          );
+        const settingsTasks = initializationTasks
+          .filter(({ name }) => name.startsWith('setting:'))
+          .map(({ name }) =>
+            context.wazuh_core.initialization.createNewTaskFromRegisteredTask(
+              name,
+            ),
+          );
+
+        const allUserTasks = [...indexPatternTasks, ...settingsTasks];
+        const tasks = tasksNames
+          ? allUserTasks.filter(({ name }) =>
+              tasksNames.some(taskName => taskName === name),
+            )
+          : allUserTasks;
+
+        logger.debug(
+          `Initialzation tasks related to user [${username}] scope [${scope}]: [${tasks
+            .map(({ name }) => name)
+            .join(', ')}]`,
+        );
+
+        const taskContext = context.wazuh_core.initialization.createRunContext(
+          'user',
+          { core: context.core, request },
+        );
+
+        logger.debug(`Running tasks for user [${username}] scope [${scope}]`);
+        const results = await Promise.all(
+          tasks.map(async task => {
+            const taskLogger = enhanceTaskLogger(logger);
+            let data;
+            try {
+              data = await task.run({
+                ...taskContext,
+                // TODO: use user selection index patterns
+                logger: taskLogger,
+                ...(task.name.includes('index-pattern:')
+                  ? {
+                      getIndexPatternID: () =>
+                        task.name /* TODO: use request parameters/body/cookies */,
+                    }
+                  : {}),
+              });
```

Comment:
```suggestion
              const taskContextWithLogger = {
                ...taskContext,
                logger: taskLogger,
              };

              if (task.name.includes('index-pattern:')) {
                taskContextWithLogger.getIndexPatternID = () => task.name; // TODO: use request parameters/body/cookies
              }

              data = await task.run(taskContextWithLogger);
```

Reply:
As we talked in a meeting, we need to review if this can be checked by the linter or define how to manage in the styleguide.

Reply:
For now, I will resolve this conversation.